lib/sdm.js

```diff
@@ -31,7 +36,28 @@ module.exports = class SDMAttachmentsService extends (
     return this.creds;
   }
 
+  async checkRepositoryType(req) {
+    const { repositoryId } = getConfigurations();
+    //check if repository is versionable
+    let repotype = cache.get(repositoryId) 
+    let isVersioned;
+    if (repotype == undefined) {
+      const token = await fetchAccessToken(
+        this.creds,
+        req.user.tokenInfo.getTokenValue()
+      );
+      const repoInfo = await getRepositoryInfo(this.creds, token);
+      isVersioned = await isRepositoryVersioned(repoInfo, repositoryId); 
+    } else { 
+      isVersioned = repotype == "versioned" ? true : false;
+    }
+    if (isVersioned) {
+      req.reject(400, "Versioned repositories are not supported");
```

Comment:
can we add this message to constant file

Reply:
Done